### ISSUE TITLE
update

### ISSUE THREAD

**Issue by bscuderi13** [pdlove/homebridge-smartthings#149]
I kept getting the email from smarthtings about needing to be on the new version which I was so I figured it was a blanket email so I ignored it. However, today i got shut down so I obviously do not have this dialed in properly. Upon investigating it appeared my direct communication definitely isnt working. I re-read the directions and saw i needed to open ports so I used the port forwarding setting "not sure if i should have used port triggering instead" I set to forward tcp udp 8000 to the raspberry pi ip. still when i run homebridge i get nothing as far as direct communication and yes i am on the new api heres a look at my config and load screen

{
    "bridge": {
        "name": "Homebridge",
        "username": "B8:27:EB:53:73:7E",
        "port": 51826,
        "pin": "031-45-154"
    },
    "description": "JSON API",
    "platforms": [
        {
            "platform": "SmartThings",
            "name": "SmartThings",
            "app_url": "https://graph-na02-useast1.api.smartthings.com:443/api/smartapps/installations/",
            "app_id": "deleted)",
            "access_token": "stuff;)",
            "update_method": "deleted",
            "direct_ip": "192.168.1.13",
            "direct_port": 8000
        }

[SmartThings] Error received type RateLimit - Please try again later
[2/17/2017, 12:03:12 AM] [SmartThings] Unknown Capabilities: []
[2/17/2017, 12:03:13 AM] Software version for Hallway Thermostat is: 5.6-7
[2/17/2017, 12:03:13 AM] Temperature unit for Hallway Thermostat is: Fahrenheit
[2/17/2017, 12:03:13 AM] Current temperature for Hallway Thermostat is: 70 F
[2/17/2017, 12:03:13 AM] Current humidity for Hallway Thermostat is: 35%
[2/17/2017, 12:03:13 AM] Target temperature for Hallway Thermostat is: 66 F
[2/17/2017, 12:03:13 AM] Away for Hallway Thermostat is: 0
[2/17/2017, 12:03:13 AM] Initializing platform accessory 'Hallway Thermostat'...
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:

    ┌────────────┐
    │ 031-45-154 │
    └────────────┘

[2/17/2017, 12:03:13 AM] Homebridge is running on port 51826.

I decided to fool around and I had missed an update from the github for smarthtings even though I think I was on a new enough version I shouldnt have been shutdown. After installing I reset up the json api and put the new app keys in my config.json file and boom it finds everything again now  although its not working im guessing i may need to delete the hub and repair with my ios app? Although Im pretty sure I will just get shutdown again cause I dont think direct is working here is my screen after updating 

[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Motion Sensor'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Office Light'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Pantry Door'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Pantry Lights'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Patio Lights'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Pool Light'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Pool Pump'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Street Light'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Switch Backlighting'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Switch Bedroom'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Switch Cabinet'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'TV Backlighting'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Water Heater'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Ainsley Kinsey'...
[2/17/2017, 12:27:10 AM] [SmartThings] Initializing platform accessory 'Ashley's iPhone'...
[2/17/2017, 12:27:11 AM] [SmartThings] Initializing platform accessory 'Brennon's iPhone'...
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:

    ┌────────────┐
    │ 031-45-154 │
    └────────────┘

[2/17/2017, 12:27:11 AM] Homebridge is running on port 51826.

**Comment by pdlove** [comment]
The port forwarding isn't needed from the internet because the hub communicates directly to the homebridge-smartthings implementation. you probably need to delete the update_method line unless you need to specify it for one of the special methods.

**Comment by bscuderi13** [comment]
okay I've removed it completely here is my smart-things portion it still isn't giving me the communicating messages after the pin. Heres the smarthings portion of my config.json is there anything missing or that should be there?

{
    "bridge": {
        "name": "Homebridge",
        "username": "B8:27:EB:53:73:7E",
        "port": 51826,
        "pin": "031-45-154"
    },
    "platforms": [
        {
            "platform": "homebridge-smartthings.SmartThings",
            "name": "SmartThings",
            "app_url": "https://graph-na02-useast1.api.smartthings.com:443/api/smartapps/installations/",
            "app_id": "stuff",
            "access_token": "stuff"
        },

.........
[2/17/2017, 9:25:38 PM] [SmartThings] Initializing platform accessory 'Ashley's                                                                                                                                                              iPhone'...
[2/17/2017, 9:25:38 PM] [SmartThings] Initializing platform accessory 'Brennon's                                                                                                                                                              iPhone'...
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:

    ┌────────────┐
    │ 031-45-154 │
    └────────────┘

[2/17/2017, 9:25:38 PM] Homebridge is running on port 51826.

**Comment by bscuderi13** [comment]
update I solved my problem... i'm still not entirely sure what was causing it but I decided to do a fresh raspbian install and then installed homebridge and smartthings from scratch on the raspberry pi. I left all extra stuff off the config.json and it worked I got all the messages that I was supposed  to. So its a mystery and I'm not really sure what wasn't happening right but somehow starting completely over worked in this case.
